src/cfnlint/rules/resources/stepfunctions/StateMachine.py

```diff
@@ -40,15 +40,15 @@ def _check_state_json(self, def_json, state_name, path):
         ]
         state_key_types = {
             'Pass': ['Result', 'ResultPath', 'Parameters'],
-            'Task': ['Resource', 'ResultPath', 'Retry', 'Catch',
+            'Task': ['Resource', 'ResultPath', 'ResultSelector', 'Retry', 'Catch',
                      'TimeoutSeconds', 'Parameters', 'HeartbeatSeconds'],
             'Map': ['MaxConcurrency', 'Iterator', 'ItemsPath', 'ResultPath',
-                    'Retry', 'Catch', 'Parameters'],
+                    'ResultSelector', 'Retry', 'Catch', 'Parameters'],
             'Choice': ['Choices', 'Default'],
             'Wait': ['Seconds', 'Timestamp', 'SecondsPath', 'TimestampPath'],
             'Succeed': [],
             'Fail': ['Cause', 'Error'],
-            'Parallel': ['Branches', 'ResultPath', 'Retry', 'Catch']
+            'Parallel': ['Branches', 'ResultPath', 'ResultSelector', 'Parameters', 'Retry', 'Catch']
```

Comment:
I don't see `Parameters` here https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-parallel-state.html in `Parallel`.  does `Parameters` apply here?

Reply:
validated `Parameters` is allowed.